test/compilable/test22665.d

```diff
@@ -0,0 +1,18 @@
+// EXTRA_FILES: imports/imp22665.d
+
+// https://issues.dlang.org/show_bug.cgi?id=22665
+
+import imports.imp22665;
+
+E foo1(E e)
+{
+    return e.A; // with qualification, it is an enum
```

Comment:
```suggestion
    return E.A; // with qualification, it is an enum
```
Did you mean for this to be upper case?  Also, why the parameter?

Reply:
Yes, and the parameter declares the `e`

Reply:
I added an E.A test.